test/UniswapV3Staker.unit.spec.ts

```diff
@@ -36,14 +31,18 @@ describe('UniswapV3Staker.unit', async () => {
   let factory: IUniswapV3Factory
   let nft: INonfungiblePositionManager
   let staker: UniswapV3Staker
+  let pool01: string
+  let pool12: string
   let subject
 
-  beforeEach('loader', async () => {
+  before('loader', async () => {
```

Comment:
I would love to know where the hardhat snapshots were getting mismatched and causing problems if you have an explanation :)

Reply:
There were two places, mintPosition was using the `nft.on('Transfer'...` event callback which was colliding between runs. Run 2 would use the callback from run 1 and things were getting really confused, thus that refactor.

I think the rest of them were small issues around promise ordering and some fire and forget methods eventually resolving during the run of another test and colliding again.